crates/turborepo-paths/src/absolute_system_path.rs

```diff
@@ -403,6 +403,16 @@ impl AbsoluteSystemPath {
         fs::read_to_string(&self.0)
     }
 
+    /// Attempts to read a file, and:
+    /// If the file does not exist it returns the default value.
+    /// For all other scenarios passes through the `read_to_string` results.
+    pub fn read_or_default(&self, default: String) -> Result<String, io::Error> {
+        match fs::read_to_string(&self.0) {
+            Err(e) if e.kind() == std::io::ErrorKind::NotFound => Ok(default),
+            read_result => read_result,
+        }
+    }
```

Comment:
This was me cleaning up this mess: https://github.com/vercel/turbo/pull/5796/commits/48117f76720ea1b8b5f967df0f15dcce3b65e887

It's intended to be a convenience because I need a JSON string at the next step, no matter what. I can improve the API a bit with generics, but that doesn't prevent a stack allocation.

```rust
pub fn read_or_default<I>(&self, default_value: I) -> Result<String, io::Error>
where I: Into<String>,
{
    match fs::read_to_string(&self.0) {
        Err(e) if e.kind() == std::io::ErrorKind::NotFound => Ok(default_value.into()),
        read_result => read_result,
    }
}
```

Reply:
Here is some incremental cleanup: https://github.com/vercel/turbo/pull/5796/commits/23dc6b755a92e56ef30948e3a908617f5f406de9

This happens twice per run, three times if link/unlink/login/logout ... I'm not super-inclined to hyper-optimize this.

Also, I _think_ that the generic trick should mean the compiler sees `"{}"` as `&'static str`?

Reply:
> Also, I think that the generic trick should mean the compiler sees "{}" as &'static str and dumps it in rodata?

Correct